boilerplates/lucia-auth/files/server/lucia-auth-handlers.ts

```diff
@@ -0,0 +1,303 @@
+    },
+  });
+}
+
+export async function luciaAuthLogoutHandler<Context extends Record<string | number | symbol, unknown>>(
+  _request: Request,
+  context: Context & { session: Session },
+): Promise<Response> {
+  const { session } = context;
+
+  if (!session) {
+    return new Response(null, {
+      status: 401,
+    });
+  }
+  await lucia.invalidateSession(session.id);
+
+  return new Response(null, {
+    status: 301,
+    headers: {
+      Location: "/login",
+      "set-cookie": lucia.createBlankSessionCookie().serialize(),
+    },
+  });
+}
+
+export async function luciaGithubLoginHandler<Context extends Record<string | number | symbol, unknown>>(
+  _request: Request,
+  _context: Context,
+): Promise<Response> {
+  const state = generateState();
+  const url = await github.createAuthorizationURL(state);
+
+  return new Response(null, {
+    status: 301,
+    headers: {
+      Location: url.toString(),
+      "set-cookie": serialize("github_oauth_state", state, {
+        path: "/",
+        secure: process.env.NODE_ENV === "production",
+        httpOnly: true,
+        maxAge: 60 * 10,
+        sameSite: "lax",
+      }),
+    },
+  });
+}
+
+export async function luciaGithubCallbackHandler<Context extends Record<string | number | symbol, unknown>>(
+  request: Request,
+  _context: Context,
+): Promise<Response> {
+  const cookies = parse(request.headers.get("cookie") || "");
+  const params = new URL(request.url).searchParams;
+  const code = params.get("code");
+  const state = params.get("state");
+  const storedState = cookies.github_oauth_state || null;
+
+  if (!code || !state || !storedState || state !== storedState) {
+    return new Response(null, {
+      status: 400,
+      headers: {
+        "content-type": "application/json",
+      },
+    });
+  }
+
+  try {
+    const tokens = await github.validateAuthorizationCode(code);
+    const githubUserResponse = await fetch("https://api.github.com/user", {
+      headers: {
+        Authorization: `Bearer ${tokens.accessToken}`,
+      },
+    });
+    const githubUser = (await githubUserResponse.json()) as GitHubUser;
+
+    const existingAccount = db
+      .prepare("SELECT * FROM oauth_accounts WHERE provider_id = ? AND provider_user_id = ?")
+      .get("github", githubUser.id) as DatabaseUser | undefined;
+
+    if (existingAccount) {
+      const session = await lucia.createSession(existingAccount.id, {});
+      return new Response(JSON.stringify({ status: "success" }), {
+        status: 301,
+        headers: {
+          Location: "/",
+          "set-cookie": lucia.createSessionCookie(session.id).serialize(),
+        },
+      });
+    }
+    const userId = generateId(15);
+
+    db.prepare("INSERT INTO users (id, username) VALUES (?, ?)").run(userId, githubUser.login);
+
+    db.prepare("INSERT INTO oauth_accounts (provider_id, provider_user_id, user_id) VALUES (?, ?, ?)").run(
+      "github",
+      githubUser.id,
+      userId,
+    );
```

Comment:
Move those queries in another file and import them.
Make use of the new [include-if-imported](https://github.com/batijs/bati/blob/c857e13d382efb4ee43400ff715f5f78656209d5/boilerplates/shared-server/files/server/vike-handler.ts#L1) instruction, it will simplify support for other databases/ORMs.

Reply:
what do you think about this https://github.com/vikejs/bati/pull/310/commits/fb94a87085b4e537ef2f25e75331f7abc5e98742 ?

Reply:
> what do you think about this [fb94a87](https://github.com/vikejs/bati/commit/fb94a87085b4e537ef2f25e75331f7abc5e98742) ?

:100: